src/ast/ast_validations.c

```diff
@@ -906,55 +906,72 @@ static AST_Validation _Validate_MERGE_Clauses(const AST *ast) {
 	return res;
 }
 
-// Validate each entity referenced in the CREATE clause.
-static AST_Validation _Validate_CREATE_Entities(const cypher_astnode_t *clause,
+// Validate each entity referenced in a single path of a CREATE clause.
+static AST_Validation _Validate_CREATE_Entities(const cypher_astnode_t *path,
 												rax *defined_aliases) {
-	const cypher_astnode_t *pattern = cypher_ast_create_get_pattern(clause);
-	// Verify that functions invoked in the CREATE pattern are valid.
-	if(_ValidateFunctionCalls(pattern, false) != AST_VALID) return AST_INVALID;
-	uint path_count = cypher_ast_pattern_npaths(pattern);
-	for(uint i = 0; i < path_count; i ++) {
-		const cypher_astnode_t *path = cypher_ast_pattern_get_path(pattern, i);
-		// Validate that inlined properties are valid.
-		if(_ValidateInlinedPropertiesOnPath(path) != AST_VALID) return AST_INVALID;
-
-		uint nelems = cypher_ast_pattern_path_nelements(path);
-		/* Visit every relationship (every odd offset) on the path to validate its alias and structure.
-		 * TODO There should also be a syntax error for redeclaring nodes, as in:
-		 * MATCH (a) CREATE (a)
-		 * But this is a no-op query, and we don't have the logic to differentiate this from a valid query like
-		 * MATCH (a) CREATE (a)-[:E]->(:B) */
-		for(uint j = 1; j < nelems; j += 2) {
-			const cypher_astnode_t *rel = cypher_ast_pattern_path_get_element(path, j);
-			const cypher_astnode_t *identifier = cypher_ast_rel_pattern_get_identifier(rel);
-			// Validate that no relation aliases are previously bound.
-			if(identifier) {
-				const char *alias = cypher_ast_identifier_get_name(identifier);
-				printf("\nDA %s\n", alias);
-				if(raxFind(defined_aliases, (unsigned char *)alias, strlen(alias)) != raxNotFound) {
-					ErrorCtx_SetError("The bound variable %s' can't be redeclared in a CREATE clause", alias);
-					return AST_INVALID;
-				}
-			}
-
-			// Validate that each relation has exactly one type.
-			uint reltype_count = cypher_ast_rel_pattern_nreltypes(rel);
-			if(reltype_count != 1) {
-				ErrorCtx_SetError("Exactly one relationship type must be specified for CREATE");
+	if(_ValidateInlinedPropertiesOnPath(path) != AST_VALID) return AST_INVALID;
+	uint nelems = cypher_ast_pattern_path_nelements(path);
+	/* Visit every relationship (every odd offset) on the path to validate its alias and structure.
+	 * TODO There should also be a syntax error for redeclaring nodes, as in:
+	 * MATCH (a) CREATE (a)
+	 * But this is a no-op query, and we don't have the logic to differentiate this from a valid query like
+	 * MATCH (a) CREATE (a)-[:E]->(:B) */
```

Comment:
is this fixed?

Reply:
No. Do you want me to address this in the scope of this PR?

Reply:
if you think is easy yes please